src/test/java/uk/gov/pay/directdebit/webhook/gocardless/resources/WebhookGoCardlessResourceOutOfOrderPaymentEventsIT.java

```diff
@@ -0,0 +1,121 @@
+@DropwizardConfig(app = DirectDebitConnectorApp.class, config = "config/test-it-config.yaml")
+public class WebhookGoCardlessResourceOutOfOrderPaymentEventsIT {
+
+    private static final GoCardlessOrganisationId GOCARDLESS_ORGANISATION_ID = GoCardlessOrganisationId.valueOf("OR123");
+    private static final GoCardlessPaymentId GOCARDLESS_PAYMENT_ID = GoCardlessPaymentId.valueOf("PM123");
+    
+    private static final Map<String, Object> PAYMENT_PAID_OUT_FAILED_OUT_OF_ORDER_WEBHOOK = Map.of(
+            "events", List.of(
+                    Map.of(
+                            "id", "EV456",
+                            "created_at", "2019-07-01T16:00:00.000Z",
+                            "resource_type", "payments",
+                            "action", "failed",
+                            "details", Map.of(
+                                    "origin", "bank",
+                                    "cause", "bank_account_closed",
+                                    "description", "The bank account for this mandate has been closed.",
+                                    "scheme", "bacs",
+                                    "reason_code", "ARUDD-B"
+                            ),
+                            "metadata", Collections.emptyMap(),
+                            "links", Map.of(
+                                    "payment", GOCARDLESS_PAYMENT_ID.toString(),
+                                    "organisation", GOCARDLESS_ORGANISATION_ID.toString()
+                            )
+                    ),
+                    Map.of(
+                            "id", "EV123",
+                            "created_at", "2019-07-01T13:00:00.000Z",
+                            "resource_type", "payments",
+                            "action", "paid_out",
+                            "details", Map.of(
+                                    "origin", "gocardless",
+                                    "cause", "payment_paid_out",
+                                    "description", "Got some dough!"
+                            ),
+                            "metadata", Collections.emptyMap(),
+                            "links", Map.of(
+                                    "payment", GOCARDLESS_PAYMENT_ID.toString(),
+                                    "organisation", GOCARDLESS_ORGANISATION_ID.toString()
+                            )
+                    )
+            )
+    );
+    
+    private static final String PAYMENT_PAID_OUT_FAILED_OUT_OF_ORDER_WEBHOOK_SIGNATURE = "dcb40e789df3aa6be84e91a567d00db269b4a107e22c49681a9d87a5e4909ad7";
+
+    @DropwizardTestContext
+    private TestContext testContext;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(10112);
+
+    private ObjectMapper objectMapper = new ObjectMapper().configure(ORDER_MAP_ENTRIES_BY_KEYS, true);
+
+    @Test
+    public void handlesOutOfOrderWebhooksByUsingLatestTimestampedEvent() throws JsonProcessingException {
```

Comment:
This tests that multiple events in the same request are processed correctly when they are out of order. Should there be another test (or included within this one) where an out of order event already exists in the db (which was what I was originally looking for before I realised that both events are in the request).

Reply:
We have something like that in `WebhookGoCardlessResourceOutOfOrderMandateEventsIT`. However, for payments, there’s only a couple of event actions that affect the state (for now), so it’s harder to do that.